crates/punktf-lib/src/transform.rs

```diff
@@ -0,0 +1,138 @@
+//! Transforms run once for each defined dotfile during the deploy process.
+//!
+//! They can either be specified for a whole profile, in which case each dotfile
+//! is transformed by them or they can be attached to a specific dotfile.
+//!
+//! The transformation takes place after the template resolving and takes the
+//! contents in a textual representation. After processing the text a new text
+//! must be returned.
+
+use std::fmt;
+
+use color_eyre::Result;
+
+/// A transform takes the contents of a dotfile, processes it and returns a new
+/// version of the content.
+///
+/// The dotfile is either the text of a resolved template or a non-template
+/// dotfile.
+pub trait Transform {
+	/// Takes a string as input, processes it and returns a new version of it.
+	///
+	/// # Errors
+	///
+	/// If any error occurs during the processing it can be returned.
+	fn transform(&self, content: String) -> Result<String>;
+}
+
+/// List of all available [`Transform`]s.
+///
+/// These can be added to a [`Profile`](`crate::profile::Profile`) or a
+/// [`Dotfile`](`crate::Dotfile`) to modify the text content.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub enum ContentTransformer {
+	/// Transformer which replaces line termination characters with either unix
+	/// style (`\n`) or windows style (`\r\b`).
+	LineTerminator(LineTerminator),
+}
+
+impl Transform for ContentTransformer {
+	fn transform(&self, content: String) -> Result<String> {
+		match self {
+			Self::LineTerminator(lt) => lt.transform(content),
+		}
+	}
+}
+
+impl fmt::Display for ContentTransformer {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+		fmt::Display::fmt(&self, f)
+	}
+}
+
+/// Transformer which replaces line termination characters with either unix
+/// style (`\n`) or windows style (`\r\b`).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
+pub enum LineTerminator {
+	/// Replaces all occurrences of `\r\n` with `\n` (unix style).
+	Lf,
```

Comment:
shouldn't that be upper case since each char stands for its own word? e.g LF and CRLF

Reply:
I thought about it but rust for example has the `Cow` struct (Copy on write) or `Arc`. I also tought about `Lf` and `CrLf` but i think thats the worst of them all.
I think your solution is probably the best one.

Reply:
Serde also allows us to set an/multiple aliases for the enum variant. So we also could allows e.g. `crlf` `CrLf` if we wanted to. I dont know it that would help or confuse people though.